android/java/org/chromium/chrome/browser/rewards/BraveRewardsBannerInfo.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.rewards;
+
+import android.content.Context;
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import androidx.annotation.IntDef;
+import androidx.annotation.VisibleForTesting;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Arrays;
+import java.util.HashMap;
+
+/**
+ * BraveRewardsNativeWorker.GetPublisherBanner json response converted to this pojo class
+ * */
+public class BraveRewardsBannerInfo implements Parcelable {
+    public static final String PUBLISHER_KEY = "publisher_key";
+    public static final String TITLE = "title";
+    public static final String NAME = "name";
+    public static final String DESCRIPTION = "description";
+    public static final String BACKGROUND = "background";
+    public static final String LOGO = "logo";
+    public static final String PROVIDER = "provider";
+    public static final String LINKS = "links";
+    public static final String STATUS = "status";
+
+    @Retention(RetentionPolicy.SOURCE)
+    @IntDef({NOT_CONNECTED, CONNECTED, VERIFIED, DISCONNECTED_NOT_VERIFIED, DISCONNECTED_VERIFIED,
+            PENDING})
+    public @interface WalletStatus {}
+
+    public static final int NOT_CONNECTED = 0;
+    public static final int CONNECTED = 1;
+    public static final int VERIFIED = 2;
+    public static final int DISCONNECTED_NOT_VERIFIED = 3;
+    public static final int DISCONNECTED_VERIFIED = 4;
+    public static final int PENDING = 5;
+
+    private String mPublisherKey;
+    private String mTitle;
+    private String mName;
+    private String mDescription;
+    private String mBackground;
+
+    private String mLogo;
+    private String mProvider;
+    private HashMap<String, String> mLinks;
+    @WalletStatus
+    private int mStatus;
+
+    protected BraveRewardsBannerInfo(Parcel in) {
+        mPublisherKey = in.readString();
+        mTitle = in.readString();
+        mName = in.readString();
+        mDescription = in.readString();
+        mBackground = in.readString();
+        mLogo = in.readString();
+        mProvider = in.readString();
+        mStatus = in.readInt();
```

Comment:
don't need `mLinks` here?

Reply:
removed parcelabale implemenetion, this class only used for converting json to pojo class.
where ever I was passing this object, Now I am sending directly string json through bundle.